src/sdk/pynni/nni/compression/torch/pruning/agp.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+import torch
+from schema import And, Optional
+from .constants import masker_dict
+from ..utils.config_validation import CompressorSchema
+from ..compressor import Pruner
+
+__all__ = ['AGP_Pruner']
+
+logger = logging.getLogger('torch pruner')
+
+class AGP_Pruner(Pruner):
```

Comment:
Suggests `AGPPruner`

Reply:
Maybe change it later, this PR is not intended to update interface.